IdentityCore/src/network/MSIDURLSessionManager.m

```diff
@@ -25,14 +25,38 @@
 #import "MSIDURLSessionDelegate.h"
 
 static MSIDURLSessionManager *s_defaultManager = nil;
+static NSTimeInterval s_timeoutIntervalForResource = 0;
 
 @implementation MSIDURLSessionManager
 
-+ (void)initialize
+- (instancetype)initWithConfiguration:(NSURLSessionConfiguration *)configuration
+                             delegate:(MSIDURLSessionDelegate *)delegate
+                        delegateQueue:(NSOperationQueue *)delegateQueue
+{
+    self = [super init];
+    if (self)
+    {
+        _configuration = configuration;
+        _session = [NSURLSession sessionWithConfiguration:configuration delegate:delegate delegateQueue:delegateQueue];
+    }
+    
+    return self;
+}
+
+- (void)dealloc
 {
-    if (self == [MSIDURLSessionManager self])
+    [_session invalidateAndCancel];
+}
+
++ (MSIDURLSessionManager *)defaultManager
+{
+    if (!s_defaultManager)
```

Comment:
Should this also be wrapped in dispatch once? Or synchronized?

Reply:
Updated with dispatch once